openedx/core/lib/extract_zip.py

```diff
@@ -0,0 +1,57 @@
+"""
+Safe version of zipfile.extractall which does not extract any files that would
+be, or symlink to a file that is, outside of the directory extracted in.
+
+Adapted from:
+http://stackoverflow.com/questions/10060069/safely-extract-zip-or-tar-using-python
+"""
+
+import logging
+from os.path import join as joinpath
+from os.path import abspath, dirname, realpath
+
+from django.conf import settings
+from django.core.exceptions import SuspiciousOperation
+
+log = logging.getLogger(__name__)
+
+
+def resolved(rpath):
+    """
+    Returns the canonical absolute path of `rpath`.
+    """
+    return realpath(abspath(rpath))
+
+
+def _is_bad_path(path, base):
+    """
+    Is (the canonical absolute path of) `path` outside `base`?
+    """
+    return not resolved(joinpath(base, path)).startswith(base)
+
+
+def safemembers(members, base):
```

Comment:
AFAIK you can have symlinks in zip files. Don't know about the rest. Even if such files don't exist, is there any issue in running the checks just to have the same interface? i.e. run the test if the interface exists and skip it otherwise.

Reply:
the link in the top of the file [explains that ZIP is safe](http://stackoverflow.com/questions/10060069/safely-extract-zip-or-tar-using-python):

> Starting with python 2.7.4, this is a non-issue for ZIP archives:
> The method zipfile.extract() prohibits the creation of files outside the sandbox